Mathlib/Data/Set/Basic.lean

```diff
@@ -2168,4 +2168,18 @@ end Disjoint
 @[simp] theorem Prop.compl_singleton (p : Prop) : ({p}ᶜ : Set Prop) = {¬p} :=
   ext fun q ↦ by simpa [@Iff.comm q] using not_iff
 
+namespace Equiv
+
+/-- Given a predicate `p : Set α → Prop`, produces an equivalence between
+  `{S : Set (Set α) // ∀ s ∈ S, p s}` and `Set {s : Set α // p s}`. -/
+protected def setSubtypeCongr {α : Type*} (p : Set α → Prop) :
+    Equiv {S : Set (Set α) // ∀ s ∈ S, p s} (Set {s : Set α // p s}) where
```

Comment:
Am I just confused? Doesn't this generalize from `Set \a` to any type?

Reply:
I'm not sure if it generalizes to 'any' type, but it definitely generalizes to set-like types.

Reply:
> Am I just confused? Doesn't this generalize from `Set \a` to any type?

Or do you mean to replace the inner `Set \a`? Then I think that's a yes. I'll generalize this soon.